web/modules/custom/joinup_core/src/JoinupRelationManager.php

```diff
@@ -123,4 +134,122 @@ public function getParentElibrary(EntityInterface $entity) {
     return $e_library;
   }
 
+  /**
+   * Retrieves all the users that have the administrator role in a group.
+   *
+   * @param \Drupal\Core\Entity\EntityInterface $entity
+   *   The group entity.
+   * @param array $state
+   *   (optional) An array of membership states to retrieve. Defaults to active.
+   *
+   * @return array
+   *   An array of users that are administrators of the entity group.
+   */
+  public function getGroupOwners(EntityInterface $entity, array $state = [OgMembershipInterface::STATE_ACTIVE]) {
+    $role_id = $entity->getEntityTypeId() . '-' . $entity->bundle() . '-administrator';
+
+    $users = [];
+    foreach ($this->getGroupMemberships($entity, $state) as $membership) {
+      $user = $membership->getUser();
+      if (!empty($user) && $membership->hasRole($role_id)) {
+        $users[$user->id()] = $user;
+      }
+    }
+
+    return array_filter($users);
+  }
+
+  /**
+   * Retrieves all the members with any role in a certain group.
+   *
+   * @param \Drupal\Core\Entity\EntityInterface $entity
+   *   The group entity.
+   * @param array $state
+   *   (optional) An array of membership states to retrieve. Defaults to active.
+   *
+   * @return array
+   *   An array of users that are members of the entity group.
+   */
+  public function getGroupUsers(EntityInterface $entity, array $state = [OgMembershipInterface::STATE_ACTIVE]) {
+    $users = array_map(function (OgMembershipInterface $membership) {
+      $user = $membership->getUser();
+      return empty($user) ? NULL : $user->id();
+    }, $this->getGroupMemberships($entity, $state));
+
+    return array_filter($users);
```

Comment:
Sorry I mean `array_reduce()` instead of `array_filter()`:

```
array_reduce($memberships, function ($users, $membership) {
  if (!empty($user = $membership->getUser())) $users[] = $user;
  return $users;
}, []);
```